packages/aws-cdk/lib/util/work-graph-builder.ts

```diff
@@ -37,27 +38,26 @@ export class WorkGraphBuilder {
    * Oof, see this parameter list
    */
   // eslint-disable-next-line max-len
-  private addAsset(parentStack: cxapi.CloudFormationStackArtifact, assetArtifact: cxapi.AssetManifestArtifact, assetManifest: AssetManifest, asset: IManifestEntry) {
+  private addAsset(parentStack: cxapi.CloudFormationStackArtifact, assetManifestArtifact: cxapi.AssetManifestArtifact, assetManifest: AssetManifest, asset: IManifestEntry) {
     // Just the artifact identifier
     const assetId = asset.id.assetId;
-    // Unique per destination where the artifact needs to go
-    const assetDestinationId = `${asset.id}`;
 
-    const buildId = `${this.idPrefix}${assetId}-build`;
-    const publishNodeId = `${this.idPrefix}${assetDestinationId}-publish`;
+    const buildId = `build-${assetId}-${contentHashAny([assetId, asset.genericSource]).substring(0, 10)}`;
```

Comment:
- Why no longer consider `this.idPrefix`? was it not needed here in the first place? 
- Why substringing? 
- Would this trigger a re upload of basically all assets everywhere? thus doubling the amount of assets in the user's account?

Reply:
> Why no longer consider `this.idPrefix`? was it not needed here in the first place?

It was an attempt to prevent false sharing between assets in subassemblies and top level assemblies. Because of the hash, this is now no longer necessary (and in fact hampers legitimate sharing).

> Why substringing?

Just to keep the id somewhat reasonable.

> Would this trigger a re upload of basically all assets everywhere? thus doubling the amount of assets in the user's account?

No effect there.